lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,433 @@
+                        if 'keys' in policy:
+                            policy.setdefault("permissions", {})["keys"] = policy["keys"]
+                            policy.pop("keys", None)
+                        if 'secrets' in policy:
+                            policy.setdefault("permissions", {})["keys"] = policy["keys"]
+                            policy.pop("secrets", None)
+                        if 'certificates' in policy:
+                            policy.setdefault("permissions", {})["keys"] = policy["keys"]
+                            policy.pop("certificates", None)
+                        if 'storage' in policy:
+                            policy.setdefault("permissions", {})["keys"] = policy["keys"]
+                            policy.pop("storage", None)
+                    self.parameters.setdefault("properties", {})["access_policies"] = access_policies
+                elif key == "enabled_for_deployment":
+                    self.parameters.setdefault("properties", {})["enabled_for_deployment"] = kwargs[key]
+                elif key == "enabled_for_disk_encryption":
+                    self.parameters.setdefault("properties", {})["enabled_for_disk_encryption"] = kwargs[key]
+                elif key == "enabled_for_template_deployment":
+                    self.parameters.setdefault("properties", {})["enabled_for_template_deployment"] = kwargs[key]
+                elif key == "enable_soft_delete":
+                    self.parameters.setdefault("properties", {})["enable_soft_delete"] = kwargs[key]
+                elif key == "create_mode":
+                    self.parameters.setdefault("properties", {})["create_mode"] = kwargs[key]
+
+        old_response = None
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(KeyVaultManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        resource_group = self.get_resource_group(self.resource_group)
+
+        if "location" not in self.parameters:
+            self.parameters["location"] = resource_group.location
+
+        old_response = self.get_keyvault()
+
+        if not old_response:
+            self.log("Key Vault instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("Key Vault instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                self.log("Need to check if Key Vault instance has to be deleted or may be updated")
+                self.to_do = Actions.Update
+
+        if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
+            self.log("Need to Create / Update the Key Vault instance")
+
+            if self.check_mode:
```

Comment:
Running this in check mode will always result in a change even when an update is not needed. You need to do check the existing object to see if it needs updating before you exit here.